src/main/java/com/hivemq/extensions/log/mqtt/message/util/MessageLogUtil.java

```diff
@@ -531,15 +535,18 @@ public static void logPubcomp(
     }
 
     private static @NotNull String getPublishAsString(
-            final @NotNull PublishPacket publishPacket, final boolean verbose) {
+            final @NotNull PublishPacket publishPacket, final boolean verbose, final boolean payload) {
         final int qos = publishPacket.getQos().getQosNumber();
         final boolean retained = publishPacket.getRetain();
-        final Optional<ByteBuffer> payload = publishPacket.getPayload();
-        final String payloadAsString = getStringFromByteBuffer(payload.orElse(null));
+        final Optional<ByteBuffer> publishPayload = publishPacket.getPayload();
+        final String payloadAsString = getStringFromByteBuffer(publishPayload.orElse(null));
```

Comment:
```suggestion
        final String payloadAsString;
        if (payload) {
            payloadAsString = getStringFromByteBuffer(publishPacket.getPayload().orElse(null));
        } else {
            payloadAsString = null;
        }
```

avoids reading the payload every time